File: src/xunit.installer.templates/MVC/VisualBasic/MVC2-VS2010/Controllers/HomeControllerFacts.vb
Imports System.Web.Mvc
Imports Xunit

Public Class HomeControllerFacts
    Public Class Index
        <Fact()> Public Sub ReturnsViewResultWithDefaultViewName()
            ' Arrange
            Dim controller = New HomeController()

            ' Act
            Dim result = controller.Index()

            ' Assert
            Dim viewResult = Assert.IsType(Of ViewResult)(result)
            Assert.Empty(viewResult.ViewName)
        End Sub

        <Fact()> Public Sub SetsViewDataWithNoModel()
            ' Arrange
            Dim controller = New HomeController()

            ' Act
            Dim result = controller.Index()

            ' Assert
            Dim viewResult = Assert.IsType(Of ViewResult)(result)
            Assert.Equal("Welcome to ASP.NET MVC!", viewResult.ViewData("Message"))
            Assert.Null(viewResult.ViewData.Model)
        End Sub
    End Class

    Public Class About
        <Fact()> Public Sub ReturnsViewResultWithDefaultViewName()
            ' Arrange
            Dim controller = New HomeController()

            ' Act
            Dim result = controller.About()

            ' Assert
            Dim viewResult = Assert.IsType(Of ViewResult)(result)
            Assert.Empty(viewResult.ViewName)
        End Sub

        <Fact()> Public Sub SetsViewDataWithNoModel()
            ' Arrange
            Dim controller = New HomeController()

            ' Act
            Dim result = controller.Index()

            ' Assert
            Dim viewResult = Assert.IsType(Of ViewResult)(result)
            Assert.Null(viewResult.ViewData.Model)
        End Sub
    End Class
End Class

File: src/xunit.installer.templates/MVC/CSharp/MVC2-VS2008/Controllers/HomeControllerFacts.cs
using System.Web.Mvc;
using Xunit;
using $mvcprojectname$.Controllers;

namespace $safeprojectname$.Controllers
{
    public class HomeControllerFacts
    {
        public class Index
        {
			[Fact]
			public void ReturnsViewResultWithDefaultViewName()
			{
				// Arrange
                var controller = new HomeController();

				// Act
                var result = controller.Index();

				// Assert
				var viewResult = Assert.IsType<ViewResult>(result);
				Assert.Empty(viewResult.ViewName);
			}

            [Fact]
            public void SetsViewDataWithNoModel()
            {
				// Arrange
                var controller = new HomeController();

                // Act
                var result = controller.Index();

                // Assert
                var viewResult = Assert.IsType<ViewResult>(result);
                Assert.Equal("Welcome to ASP.NET MVC!", viewResult.ViewData["Message"]);
                Assert.Null(viewResult.ViewData.Model);
            }
        }

        public class About
        {
			[Fact]
			public void ReturnsViewResultWithDefaultViewName()
			{
				// Arrange
                var controller = new HomeController();

				// Act
                var result = controller.About();

				// Assert
				var viewResult = Assert.IsType<ViewResult>(result);
				Assert.Empty(viewResult.ViewName);
			}

            [Fact]
            public void SetsViewDataWithNoModel()
            {
				// Arrange
                var controller = new HomeController();

				// Act
                var result = controller.About();

                // Assert
                var viewResult = Assert.IsType<ViewResult>(result);
                Assert.Null(viewResult.ViewData.Model);
            }
        }
    }
}